evadb/executor/create_function_executor.py

```diff
@@ -422,7 +457,10 @@ def handle_forecasting_function(self):
             FunctionMetadataCatalogEntry("library", library),
         ]
 
-        os.environ.pop("CUDA_VISIBLE_DEVICES", None)
+        if save_old_cuda_env is not None:
+            os.environ["CUDA_VISIBLE_DEVICES"] = save_old_cuda_env
+        else:
+            os.environ.pop("CUDA_VISIBLE_DEVICES", None)
```

Comment:
Please check https://stackoverflow.com/questions/2059482/temporarily-modify-the-current-processs-environment for using context manager to modify the environment variable.